reactioncommerce/api-plugin-settings
# feat: move source code and enable semantic-release

This plugin replaces the internal `settings` service. 

Once this PR is merged and this plugin is published to `npm`, there will be a related PR in `reaction` to swap out this plugin for the existing version.

## Testing

- see that the plugin is loaded upon startup in Reaction (use the `systemInformation` query)
- use the settings queries and mutations through the GraphQL API and verify they are working

Example test queries/mutations:
```
query {
  shopSettings(shopId: "set-shop-id"){
    canSellVariantWithoutInventory
    sitemapRefreshPeriod
  }
}

mutation {
  updateShopSettings(input: {shopId: "set-shop-id", settingsUpdates: {canSellVariantWithoutInventory: true}}){
    shopSettings {
      canSellVariantWithoutInventory
      sitemapRefreshPeriod
    }
  }
}
```

Issue https://github.com/reactioncommerce/reaction/issues/6178

Signed-off-by: Maria Paktiti <maria.paktiti@gmail.com>
